src/main/java/com/cloudant/client/internal/URIBase.java

```diff
@@ -0,0 +1,19 @@
+package com.cloudant.client.internal;
+
+import java.net.URI;
+
+/**
+ * Contains methods for building a base URI to be used with a Cloudant account.
+ * This class is extended by DatabaseURIHelper to build URI's for Cloudant databases.
+ */
+public class URIBase extends URIBaseMethods<URIBase> {
```

Comment:
Although you have a comment here (which is a good thing!), Rich had to explain to me how these various URI classes fit together, especially the reasoning behind the type parameters and fluent style `returnThis`.

Can we please have a bit more javadoc commenting for each of the classes involved, explaining how they relate to each other.